app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java

```diff
@@ -337,50 +344,43 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     @Override
     public boolean dispatchKeyEvent(KeyEvent event) {
-
-        // only do this once per key press (otherwise the action is done twice)
         if (event.getAction() == KeyEvent.ACTION_DOWN) {
-            switch (event.getKeyCode()) {
+            int code = event.getKeyCode();
+            switch (code) {
                 case KeyEvent.KEYCODE_PAGE_UP:
-                    scroll(true, screenScrollingPercent, smoothScrolling);
-                    return true;
-
                 case KeyEvent.KEYCODE_PAGE_DOWN:
-                    scroll(false, screenScrollingPercent, smoothScrolling);
-                    return true;
-
-                case KeyEvent.KEYCODE_CAMERA:
-                    disableTouch = !disableTouch;
-                    if (disableTouch) {
-                        Toast.makeText(this, "Touch screen disabled", Toast.LENGTH_SHORT).show();
-                    } else {
-                        Toast.makeText(this, "Touch screen enabled", Toast.LENGTH_SHORT).show();
-                    }
-                    Log.d(TAG, "toggling touch screen, now disableTouch is " + disableTouch);
+                    scroll(code == KeyEvent.KEYCODE_PAGE_UP, screenScrollingPercent, smoothScrolling);
                     return true;
 
                 case KeyEvent.KEYCODE_VOLUME_UP:
-                    if (volumeButtonsScrolling) {
-                        scroll(true, screenScrollingPercent, smoothScrolling);
-                        return true;
-                    } else {
-                        return super.dispatchKeyEvent(event);
-                    }
-
                 case KeyEvent.KEYCODE_VOLUME_DOWN:
                     if (volumeButtonsScrolling) {
-                        scroll(false, screenScrollingPercent, smoothScrolling);
+                        scroll(code == KeyEvent.KEYCODE_VOLUME_UP, screenScrollingPercent, smoothScrolling);
                         return true;
-                    } else {
-                        return super.dispatchKeyEvent(event);
                     }
+                    break;
+            }
+
+            if(event.getKeyCode() == disableTouchKeyCode && (disableTouch || disableTouchOptionEnabled)) {
```

Comment:
I tried a different key for the no-touch option and, since I only have four key on my device, ended up with using the KEYCODE_PAGE_DOWN. Using such key, however, is already catched by scrolling down. Maybe we should move the no-touch action before the scrolling action?

Reply:
Good point. Done.